<launch>
  <!-- Launch arguments -->
  <arg name="car_name"         default="car"/>
  <arg name="mode_real"        default="false"/>
  <arg name="use_modified_map" default="false"/>

  <!--
  Serve AMCL with an altered map for stress testing the localization against dynamic or cluttered environments.
  To use this feature you must create both an altered map and point the config file (map_modified.yaml) to it.
  When using an altered map it's recommended to disable short range measurements in the simulated laser.
  -->
  <group if="$(arg use_modified_map)">
    <node pkg="map_server" type="map_server" name="map_modified_server"
          args="map:=/map_modified $(find amcl)/maps/modified.yaml"/>
  </group>

  <!-- AMCL -->
  <group ns="$(arg car_name)">
    <!-- Real parameters -->
    <group if="$(arg mode_real)">
      <rosparam file="$(find amcl)/config/real.yaml" command="load"/>
    </group>

    <!-- Sim parameters -->
    <group unless="$(arg mode_real)">
      <rosparam file="$(find amcl)/config/sim.yaml" command="load"/>
    </group>

    <!-- Node -->
    <node pkg="amcl" type="amcl" name="localizer" output="screen">
      <param name="amcl/update_rate"              value="50.0"/>
      <param name="amcl/use_modified_map"         value="$(arg use_modified_map)"/>
      <param name="node_names/amcl"               value="localizer"/>
      <param name="node_names/drive"              value="vesc"/>
      <param name="node_names/sensor"             value="laser"/>
      <param name="frame_ids/map"                 value="map"/>
      <param name="frame_ids/car_base"            value="$(arg car_name)/base_link"/>
      <param name="frame_ids/car_wheel_back_left" value="$(arg car_name)/back_left/wheel_link"/>
    </node>
  </group>
</launch>
